### PR TITLE
Set const for array passed or returned from C/C++ API

### PR DESCRIPTION
**Pull request by cjvolzka**
* Set parameters returns of arrays like `shape`, `strides` and some TensorList pointers as `const`
* Implements https://github.com/onnx/onnx-mlir/issues/2075

### CODE REVIEW

File: src/Runtime/OMResize.inc [hunk 1]
```diff
@@ -1,3 +1,17 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------- OMTensorList.cpp - OMTensor C/C++ Implementation ----------===//
```
Comment (tungld) [inline src/Runtime/OMResize.inc:5]: Typos: `OMTensorList` => `OMResize`. Please change the content of this comment accordingly.

**Review by tungld — APPROVED** [review]
LGTM. Thanks!

**Comment by cjvolzka** [comment]
@jenkins-droid test this please

**Comment by AlexandreEichenberger** [comment]
@cjvolzka see that you kept the `OMTensor **` for the lists... that seems very sensible.

**Pull request merged**